src/raft.rs

```diff
@@ -1939,6 +1945,7 @@ impl<T: Storage> Raft<T> {
             .maybe_append(m.index, m.log_term, m.commit, &m.entries)
         {
             to_send.set_index(last_idx);
+            to_send.set_commit(self.raft_log.committed);
             self.send(to_send);
```

Comment:
How about moving these two lines and 1964,1965 lines outside this conditional branch?